code/game/objects/items/stacks/packagewrap.dm

```diff
@@ -0,0 +1,207 @@
+		/obj/item/weapon/gift,
+		/obj/item/weapon/winter_gift,
+		/obj/item/weapon/evidencebag,
+		/obj/item/weapon/legcuffs/bolas,
+		/obj/item/weapon/storage
+		)
+
+	var/list/wrappable_big_stuff = list(
+		/obj/structure/closet,
+		/obj/structure/vendomatpack,
+		/obj/structure/stackopacks
+		)
+
+/obj/item/stack/package_wrap/afterattack(var/attacked, mob/user as mob)
+	if(ishuman(attacked))
+		try_wrap_human(attacked,user)
+		return
+	if(!istype(attacked,/obj)) return
+	var/obj/target = attacked
+	if(is_type_in_list(target, cannot_wrap))
+		return
+	if(target.anchored)
+		return
+	if(target in user)
+		return
+
+	user.attack_log += text("\[[time_stamp()]\] <font color='blue'>Has used [src.name] on \ref[target]</font>")
+	target.add_fingerprint(usr)
+	src.add_fingerprint(usr)
+
+	if(istype(target, /obj/item) && smallpath)
+		if (amount >= 1)
+			var/obj/item/I = target
+			var/obj/item/P = new smallpath(get_turf(target.loc),target,round(I.w_class))
+			if(!istype(target.loc, /turf))
+				if(user.client)
+					user.client.screen -= target
+			target.forceMove(P)
+			P.add_fingerprint(user)
+			use(1)
+		else
+			user << "<span class='warning'>You need more paper!</span>"
+	else if(is_type_in_list(target,wrappable_big_stuff) && bigpath)
+		if(istype(target,/obj/structure/closet))
+			var/obj/structure/closet/C = target
+			if(C.opened) return
+		if(amount >= 3)
+			var/obj/item/P = new bigpath(get_turf(target.loc),target)
+			target.forceMove(P)
+			P.add_fingerprint(usr)
+			use(3)
+		else
+			user << "<span class='warning'>You need more paper!</span>"
+	else
+		user << "<span class='warning'>[src] isn't useful for wrapping [target].</span>"
+	return
+
+/obj/item/stack/package_wrap/proc/try_wrap_human(var/mob/living/carbon/human/H, mob/user as mob)
+	if(!manpath) return
+	if(amount >= 2)
+		H.visible_message("<span class='danger'>[user] is trying to wrap up [H]!</span>")
+		if(do_after(user,H,human_wrap_speed))
+			var/obj/present = new manpath(get_turf(H))
+			if (H.client)
+				H.client.perspective = EYE_PERSPECTIVE
+				H.client.eye = present
+			H.loc = present
+			H.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has been wrapped with [src.name]  by [user.name] ([user.ckey])</font>")
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Used the [src.name] to wrap [H.name] ([H.ckey])</font>")
+			if(!iscarbon(user))
+				H.LAssailant = null
+			else
+				H.LAssailant = user
+			log_attack("<font color='red'>[user.name] ([user.ckey]) used the [src.name] to wrap [H.name] ([H.ckey])</font>")
+			use(2)
+	else
+		user << "<span class='warning'>You need more paper!</span>"
+
+/obj/item/stack/package_wrap/gift //For more details, see gift_wrappaper.dm
+	name = "gift wrap"
+	desc = "A festive wrap for hand-delivered presents. Not compatible with mail."
+	icon_state = "wrap_paper"
+	smallpath = /obj/item/weapon/gift
+	bigpath = null
+	manpath = /obj/structure/strange_present
+
+/obj/structure/bigDelivery

```

Comment:
With how similar these are to the smaller ones, and with how little the `/obj/structure` actually does; make these subtype of the smaller ones, and make them behave like they are now (can't be picked up, `density`...)